packages/validation/test/parseValues/parseField.spec.ts

```diff
@@ -165,6 +167,34 @@ describe('Parse Values - parseFieldValue', () => {
 				expect(parseFieldValue('     !@#$%^&* ()_+      ', fieldStringNoRestriction).success).true;
 				expect(parseFieldValue('     !@#$%^&* ()_+      ', fieldStringNoRestriction).data).equals('!@#$%^&* ()_+');
 			});
+			it('Successfuly parses strings, with conditional restrictions', () => {
+				const value = 'any random string value!!!';
+				const result = parseFieldValue(value, fieldStringConditionalExists);
+				expect(result.success).true;
+				expect(result.data).equal(value);
+
+				expect(parseFieldValue('    123', fieldStringConditionalExists).success).true;
+				expect(parseFieldValue('    123', fieldStringConditionalExists).data).equals('123');
+				expect(parseFieldValue('false   ', fieldStringConditionalExists).success).true;
+				expect(parseFieldValue('false   ', fieldStringConditionalExists).data).equals('false');
+				expect(parseFieldValue('     !@#$%^&* ()_+      ', fieldStringConditionalExists).success).true;
+				expect(parseFieldValue('     !@#$%^&* ()_+      ', fieldStringConditionalExists).data).equals('!@#$%^&* ()_+');
+			});
+			it('Successfuly parses strings, with conditional restrictions without then or else', () => {
+				const value = 'any random string value!!!';
+				const result = parseFieldValue(value, fieldStringConditionalExistsWithouthThenElse);
```

Comment:
Testing parsing any string using a Schema Field Definition having an`if` restriction without `then/else`. This is not a functional Schema definition however to cover every scenario possible we have to test it.

Reply:
Is the expected behaviour that an if without then/else just never applies restrictions?

Should we consider adding validation to lectern dictionaries to require an if to have a then or else (at least one of the two)?

Reply:
If not then/else is found in the schema definition the expected behaviour is do not apply the restriction. Based on test [resolveFieldRestrictions.spec.ts](https://github.com/overture-stack/lectern/blob/develop/packages/validation/test/validateField/resolveFieldRestrictions.spec.ts#L106)

Reply:
I'm open to add a validation to require at least one between then/else. Although it could be added in a separate PR